websocket_managed_conn.go

```diff
@@ -16,6 +16,38 @@ import (
 	"github.com/slack-go/slack/internal/timex"
 )
 
+// UnmappedError represents error occurred when there is no mapping between given event name
+// and corresponding Go struct.
+type UnmappedError struct {
+	eventType string
+	rawEvent  json.RawMessage
+	ctxMsg    string
+}
+
+// NewUnmappedError returns new UnmappedError instance.
+func NewUnmappedError(ctxMsg, eventType string, raw json.RawMessage) *UnmappedError {
+	return &UnmappedError{
+		ctxMsg:    ctxMsg,
+		eventType: eventType,
+		rawEvent:  raw,
+	}
+}
+
+// Error returns human-readable error message.
+func (u UnmappedError) Error() string {
+	return fmt.Sprintf("%s: Received unmapped event %q", u.ctxMsg, u.eventType)
+}
+
+// EventType returns event type name.
+func (u UnmappedError) EventType() string {
+	return u.eventType
+}
+
+// RawEvent returns raw event body.
+func (u UnmappedError) RawEvent() json.RawMessage {
+	return u.rawEvent
+}
```

Comment:
It seems to me that these fields could be exported. Qhat do you think?

Reply:
Yep, in this case it makes. I automatically went with the method style to have more control in the future. For example:
```go
type Event struct {
	eType *string
}

func (e *Event) EventType() string {
	if e.eType == nil {
		return "unknown"
	}
	return *e.eType
}
```

but it's not the case here, so I changed as suggested 👍